tests/protocols/test_http.py

```diff
@@ -801,3 +812,75 @@ def send_fragmented_req(path):
     assert bad_response != response[: len(bad_response)]
     server.should_exit = True
     t.join()
+
+
+@pytest.mark.anyio
+async def test_huge_headers_h11protocol_failure():
+    app = Response("Hello, world", media_type="text/plain")
+
+    protocol = get_connected_protocol(app, H11Protocol)
+    # Huge headers make h11 fail in it's default config
+    # h11 sends back a 400 in this case
+    protocol.data_received(GET_REQUEST_HUGE_HEADERS[0])
+    assert b"HTTP/1.1 400 Bad Request" in protocol.transport.buffer
+    assert b"Connection: close" in protocol.transport.buffer
+    assert b"Invalid HTTP request received." in protocol.transport.buffer
+
+
+@pytest.mark.anyio
+@pytest.mark.skipif(HttpToolsProtocol is None, reason="httptools is not installed")
```

Comment:
is it really necessary to have this skipif ?

Reply:
Hmmm... No? I'll check in an hour.

I've added so the test logic doesn't run when httptools not installed.

Reply:
but the test suite has httptools installed no ? or we should skip on windows ?

Reply:
We usually take into consideration the setup of ppl that run our test suite locally (there are some conditionals related to this idea somewhere I think... I'll check soon).

Reply:
ok ok , one of these days where I was off :) fine by me

Reply:
I think it was introduced 4 years ago 👀